tests/playbooks/roles/install-k3s/defaults/main.yaml

```diff
@@ -8,4 +8,6 @@ sg_name: "k3s_sg"
 keypair_name: "k3s_keypair"
 image_url: "https://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64.img"
 image_name: "ubuntu-focal"
-master_port_name: "k3s_master"
\ No newline at end of file
+master_port_name: "k3s_master"
+k3s_master_cloud_provider_config: "--disable-cloud-controller --kubelet-arg='cloud-provider=external'"
```

Comment:
> make k3s install flag (external cloud ) optional

@jichenjc This means we use intree cloud provider?  intree cloud providers will be removed in next release or so , so the job fails when the provider is removed from the repo right? Or I misundersood the intent? Please clarify

Reply:
@ramineni please see above discussion with @lingxiankong 
I started up with OCCM actually..

Reply:
@jichenjc Ya, I checked , Thats what I would like to confirm, not to rely on CCM means intree? Jobs would fail in that case if removed?
@lingxiankong ^^

Reply:
it takes a few more steps than current way
I guess another method is to keep this running until in-tree really removed
should be not that hard :)

Reply:
@jichenjc I dont think thats good idea to use deprecated ones in jobs

Reply:
but actually we run it for long time, don't we? 
at least it makes the first run simple (no OCCM involvement from my test so far ..)

Reply:
The prerequisite for cinder csi driver is we need Kubernetes Cluster on Openstack, the only supported/recommended way is using external provider right. We dont need to build the OCCM, but we can directly pull from docker repo the latest image. wdyt?
@lingxiankong wdyt?

Reply:
that's another option but I don't see too much difference 
after all, we have docker registry, code repo down etc ,all set for CSI already
build OCCM or pull is not a big deal

Reply:
@ramineni @jichenjc Very good discussion,  my original thought is to remove the dependency of occm (especially the image build) from cinder-csi related tests, I'm not sure if the cloud controller is now a required component for a k8s cluster?

Reply:
I knew it's gonna remove the in-tree provider soon (1.23 or 1.24? ) 
I can go with a w/ OCCM but no build image (just pull latest) approach if that's an acceptable way

Reply:
@jichenjc sounds good to me